TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Users/EditUser/Deactivate.cshtml.cs

```diff
@@ -0,0 +1,180 @@
+            // or choosing not to upload evidence (check for UploadEvidence != true because if
+            // UploadEvidence somehow got set to null we still want to delete the file)
+            if (EvidenceFileId.HasValue && (EvidenceFile is not null || UploadEvidence != true))
+            {
+                await fileService.DeleteFileAsync(EvidenceFileId.Value);
+            }
+
+            // Upload the file even if the rest of the form is invalid
+            // otherwise the user will have to re-upload every time they re-submit
+            if (UploadEvidence == true)
+            {
+                // Upload the file and set the display fields
+                if (EvidenceFile is not null)
+                {
+                    using var stream = EvidenceFile.OpenReadStream();
+                    var fileId = await fileService.UploadFileAsync(stream, EvidenceFile.ContentType);
+                    EvidenceFileName = EvidenceFile?.FileName;
+                    EvidenceFileSizeDescription = EvidenceFile?.Length.Bytes().Humanize();
+                    UploadedEvidenceFileUrl = await fileService.GetFileUrlAsync(fileId, DeactivateDefaults.FileUrlExpiry);
+                    EvidenceFileId = fileId;
+                }
+            }
+
+            if (!ModelState.IsValid)
+            {
+                return this.PageWithErrors();
+            }
+
+            _user.Active = false;
+
+            await dbContext.AddEventAndBroadcastAsync(new UserDeactivatedEvent
+            {
+                EventId = Guid.NewGuid(),
+                User = EventModels.User.FromModel(_user),
+                RaisedBy = User.GetUserId(),
+                CreatedUtc = clock.UtcNow,
+                AdditionalReason = (HasAdditionalReason ?? false) ? AdditionalReasonDetail : null,
+                MoreInformation = (HasMoreInformation ?? false) ? MoreInformationDetail : null,
+                EvidenceFileId = EvidenceFileId,
+            });
+
+            await dbContext.SaveChangesAsync();
+            TempData.SetFlashSuccess(message: $"{_user.Name}\u2019s account has been deactivated.");
+
+            return Redirect(linkGenerator.Users());
+        }
+
+        public override async Task OnPageHandlerExecutionAsync(PageHandlerExecutingContext context, PageHandlerExecutionDelegate next)
+        {
+            _user = await dbContext.Users.SingleOrDefaultAsync(u => u.UserId == UserId);
+
+            if (_user is null)
+            {
+                context.Result = NotFound();
+                return;
+            }
+
+            await next();
+        }
+
+        public async Task<IActionResult> OnGetCancelAsync(Guid? evidenceFileId)
+        {
+            // If the user cancels after having uploaded a file but before submitting,
+            // we want to delete the file to clean up after ourselves.
+            // I've made it a GET rather than a POST as yes it does delete a file, but
```

Comment:
This might cause problems with browsers and setups that pre-fetch links; making it a POST is probably safer.